extensions/nanoarrow_ipc/src/nanoarrow/nanoarrow_ipc.c

```diff
@@ -785,5 +876,261 @@ ArrowErrorCode ArrowIpcReaderDecode(struct ArrowIpcReader* reader,
       return EINVAL;
   }
 
+  private_data->last_message = message_header;
+  return NANOARROW_OK;
+}
+
+ArrowErrorCode ArrowIpcReaderGetSchema(struct ArrowIpcReader* reader,
+                                       struct ArrowSchema* out,
+                                       struct ArrowError* error) {
+  struct ArrowIpcReaderPrivate* private_data =
+      (struct ArrowIpcReaderPrivate*)reader->private_data;
+
+  if (private_data->schema.release == NULL) {
+    ArrowErrorSet(error, "reader does not contain a valid schema");
+    return EINVAL;
+  }
+
+  ArrowSchemaMove(&private_data->schema, out);
```

Comment:
Do we really want Move here, or a copy?

Reply:
It does break expectations for Get though; usually you'd expect that to be idempotent